.github/workflows/tests.yml

```diff
@@ -14,18 +14,20 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        resolver: [lts-16, lts-15, lts-14]
+        resolver: [lts-16, lts-15, lts-14, lts-12]
+        yaml: ['stack.yaml', 'stack-master.yaml']
 
     steps:
       - name: Clone project
         uses: actions/checkout@v2
 
       - name: Cache dependencies
-        uses: actions/cache@v1
+        uses: actions/cache@v2
         with:
           path: ~/.stack
-          key: ${{ runner.os }}-${{ matrix.resolver }}-${{ hashFiles('stack.yaml') }}
+          key: ${{ runner.os }}-${{ matrix.resolver }}-${{ hashFiles(matrix.yaml) }}
```

Comment:
I still believe `'*.cabal'` should be added here, as otherwise stale caches with missing deps will be used whenever an additional dependency is added to the cabal file.

Reply:
@liskin I have added `xmonad-contrib.cabal` as an additional hash file.